src/store/modules/chat/index.js

```diff
@@ -712,55 +714,61 @@ const actions = {
    * @param {string} replyToId Optional
    * @returns {Promise}
    */
-  sendMessage({ commit, rootState }, { message, recipientId, replyToId }) {
-    const messageObject = createMessage({
-      message,
-      recipientId,
-      senderId: rootState.address,
-      replyToId
-    })
-
-    commit('pushMessage', {
-      message: messageObject,
-      userId: rootState.address
-    })
+  async sendMessage({ commit, rootState }, { message, recipientId, replyToId }) {
+    let id;
+    try {
+      const type = replyToId ? MessageType.RICH_CONTENT_MESSAGE : MessageType.BASIC_ENCRYPTED_MESSAGE
 
-    const type = replyToId ? MessageType.RICH_CONTENT_MESSAGE : MessageType.BASIC_ENCRYPTED_MESSAGE
-    const messageAsset = replyToId
-      ? replyMessageAsset({
+      const messageAsset = replyToId
+        ? replyMessageAsset({
           replyToId,
           replyMessage: message
         })
-      : message
+        : message
 
-    return queueMessage(messageAsset, recipientId, type)
-      .then((res) => {
-        // @todo this check must be performed on the server
-        if (!res.success) {
-          throw new Error(i18n.global.t('chats.message_rejected'))
-        }
 
-        // update `message.status` to 'REGISTERED'
-        // and `message.id` with `realId` from server
-        commit('updateMessage', {
-          id: messageObject.id,
-          realId: res.transactionId,
-          status: TS.REGISTERED, // not confirmed yet, wait to be stored in the blockchain (optional line)
-          partnerId: recipientId
-        })
+      const signedTransaction = await admApi.signChatMessageTransaction({
+        to: recipientId,
+        message: messageAsset,
+        type,
+      })
 
-        return res
+      id = adamant.getTransactionId(signedTransaction)
+
+      const messageObject = createMessage({
+        id,
+        message,
+        recipientId,
+        senderId: rootState.address,
+        replyToId
       })
-      .catch((err) => {
-        // update `message.status` to 'REJECTED'
+
+      commit('pushMessage', {
+        message: messageObject,
+        userId: rootState.address,
+      })
+
+      const transaction = await queueSignedMessage(signedTransaction)
+
+      if (!transaction.success) {
+        throw new Error(i18n.global.t('chats.message_rejected'))
+      }
+
+      commit('updateMessage', {
+        id,
+        status: TS.REGISTERED,
+        partnerId: recipientId
+      })
+    } catch (error) {
+
+      if (id) {
         commit('updateMessage', {
-          id: messageObject.id,
+          id,
           status: TS.REJECTED,
-          partnerId: recipientId
```

Comment:
As I see, `updateMessage ` mutation still requires `partnerId` property, but you removed it